srcalc/src/main/java/gov/va/med/srcalc/vista/vistalink/VistaLinkUtil.java

```diff
@@ -10,26 +17,104 @@
  */
 public class VistaLinkUtil
 {
+    private static final Logger LOGGER = LoggerFactory.getLogger(VistaLinkUtil.class);
+    
+    private static final IPrimaryStationRules STATION_RULES;
+    
+    // Eagerly get IPrimaryStationRules implementation to find configuration issues early.
+    static
+    {
+        try
+        {
+            STATION_RULES = PrimaryStationRulesFactory.getPrimaryStationRulesImplementation();
+        }
+        catch (final ReflectiveOperationException e)
+        {
+            // getPrimaryStationRulesImplementation() does not document its exceptions,
+            // so at least be as specific as we can.
+            throw new ConfigurationException(
+                    "Could not get configured IPrimaryStationRules implementation.", e);
+        }
+    }
+    
     private VistaLinkUtil()
     {
         // No construction.
     }
-
+    
     /**
-     * Returns true if the given division is known to VistALink, false otherwise.
+     * Tries to load the VistALink ConnectionFactory for the given division by explicitly
+     * reading the configuration file.
      */
-    public static boolean isDivisionKnown(final String division)
+    private static void tryForceLoad(final String division)
     {
+        // The below technique is tighter coupling with VistALink than I would like, but
+        // I couldn't find a better way. - David Tombs, 31-Aug-2015
         try
         {
-            InstitutionMappingDelegate.getJndiConnectorNameForInstitution(division);
+            final String station = STATION_RULES.getPrimaryStationLookupString(division);
+            final String configuredJndiName =
+                    ConfigurationReader.getJndiNameForStationNumber(station);
+            LOGGER.trace(
+                    "Got JNDI name {} for division {} from configuration",
+                    configuredJndiName, division);
+            if (configuredJndiName != null)
+            {
+                final InitialContext ic = new InitialContext();
+                ic.lookup(configuredJndiName);
+            }
         }
-        catch (InstitutionMapNotInitializedException | InstitutionMappingNotFoundException e)
+        catch (final Exception ex)
         {
-            return false;
+            LOGGER.info(
+                    "Swallowing Exception while trying to force-load division {}",
+                    division,
+                    ex);
```

Comment:
Never mind that. I just looked at the SLF4J FAQ about adding messages to exceptions.